pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/text/LuceneIndexSearcherReferenceManager.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.segment.index.readers.text;
+
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.SearcherManager;
+
+
+/**
+ * A reference manager to help release the Lucene text index reader/searcher.
+ * This is used only in the context of realtime search with Lucene. To avoid
+ * introducing a release() API on the index reader interface, this wrapper is
+ * created over SearcherManager and returned to the caller. The caller can
+ * then appropriately call the release once the searcher associated with
+ * SearcherManager is no longer needed.
+ */
+public class LuceneIndexSearcherReferenceManager {
+
+  private final SearcherManager _searcherManager;
+  private final IndexSearcher _indexSearcher;
+
+  public LuceneIndexSearcherReferenceManager(SearcherManager searcherManager, IndexSearcher indexSearcher) {
+    _searcherManager = searcherManager;
+    _indexSearcher = indexSearcher;
+  }
+
+  public void release() {
+    try {
+      // Only used for realtime search with Lucene
+      if (_searcherManager != null) {
+        _searcherManager.release(_indexSearcher);
+      }
+    } catch (Exception e) {
+      throw new RuntimeException("Error: failed while releasing a previously acquired searcher. " + e);
```

Comment:
Please remove the word "Error". You can log an error log. Also, e.getMessage()?

Reply:
Done